Sources/MapboxDirections/RouteStep.swift

```diff
@@ -575,7 +619,27 @@ open class RouteStep: Codable {
         typicalTravelTime = try container.decodeIfPresent(TimeInterval.self, forKey: .typicalTravelTime)
         
         transportType = try container.decode(TransportType.self, forKey: .transportType)
-        intersections = try container.decodeIfPresent([Intersection].self, forKey: .intersections)
+        
+        administrativeRegionIndicesByIntersection = try container.decodeIfPresent([AdministrativeAreaIndex].self,
+                                                                                  forKey: .intersections)?.map { $0.administrativeRegionIndex }
+        var rawIntersections = try container.decodeIfPresent([Intersection].self, forKey: .intersections)
+        
+        // Updating `Intersection.regionCode` since we removed it's `admin_index` for convenience
+        if let administrativeRegions = administrativeRegions,
+           rawIntersections != nil,
+           let rawAdminIndicies = administrativeRegionIndicesByIntersection {
+            for index in 0..<rawIntersections!.count {
```

Comment:
Can we prevent force-unwrapping usage here?

Reply:
We cannot use `let rawIntersections = rawIntersections` construct here because we need to mutate contents of this array later in the code.